src/g2_getfld.c

```diff
@@ -338,53 +340,53 @@ g2_getfld(unsigned char *cgrib, g2int ifldnum, g2int unpack, g2int expand,
         if (isecnum == 7 && numfld == ifldnum && unpack)
         {
             iofst = iofst - 40;       /* reset offset to beginning of section */
-            jerr = g2_unpack7(cgrib, &iofst, lgfld->igdtnum, lgfld->igdtmpl, 
-                            lgfld->idrtnum, lgfld->idrtmpl, lgfld->ndpts, 
-                            &lgfld->fld);
-            if (jerr == 0)
+            if ((jerr = g2_unpack7(cgrib, &iofst, lgfld->igdtnum, lgfld->igdtmpl,
+                                   lgfld->idrtnum, lgfld->idrtmpl, lgfld->ndpts,
+                                   &lgfld->fld)))
+            {
+                printf("g2_getfld: return from g2_unpack7 = %d \n", (int)jerr);
+                g2_free(lgfld);
+                ierr = 14;
+                return ierr;
+            }
+
+            have7 = 1;
+            
+            /*  If bitmap is used with this field,  expand data field */
+            /*  to grid, if possible. */
+            if (lgfld->ibmap != 255 && lgfld->bmap != 0)
             {
-                have7 = 1;
-                /*  If bitmap is used with this field,  expand data field */
-                /*  to grid, if possible. */
-                if (lgfld->ibmap != 255 && lgfld->bmap != 0)
+                if (expand == 1)
                 {
-                    if (expand == 1)
+                    n = 0;
+                    newfld = calloc(lgfld->ngrdpts, sizeof(g2float));
```

Comment:
should newfld have a free() ? i see it's allocated but not free'd

Reply:
This should be fine. C allow this.

Reply:
Indeed it must be freed! But look further in the code:

```
               if (expand == 1)
                {
                    n = 0;
                    newfld = calloc(lgfld->ngrdpts, sizeof(g2float));
                    for (j = 0; j < lgfld->ngrdpts; j++)
                    {
                        if (lgfld->bmap[j] == 1)
                            newfld[j] = lgfld->fld[n++];
                    }
                    free(lgfld->fld);
                    lgfld->fld = newfld;
                    lgfld->expanded = 1;
```
So the address is assigned to the fld member of the struct pointed to by lgfld. Later this is freed with a g2_free(), which frees all the memory allocated to any member of this struct.